File: omnes.Web/Modules/ServerTypes/Parametros/TiposDocumentoRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TiposDocumentoRow {
    IdTipoDocumento?: number;
    NombreTipoDocumento?: string;
    Baja?: boolean;
    Sigla?: string;
}

export abstract class TiposDocumentoRow {
    static readonly idProperty = 'IdTipoDocumento';
    static readonly nameProperty = 'NombreTipoDocumento';
    static readonly localTextPrefix = 'Parametros.TiposDocumento';
    static readonly deletePermission = 'SuperAdmin';
    static readonly insertPermission = 'SuperAdmin';
    static readonly readPermission = 'SuperAdmin';
    static readonly updatePermission = 'SuperAdmin';

    static readonly Fields = fieldsProxy<TiposDocumentoRow>();
}

File: omnes.Web/Modules/Parametros/TiposDocumento/TiposDocumentoRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace omnes.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("TiposDocumento")]
[DisplayName("Tipos Documento"), InstanceName("Tipos Documento")]
[ReadPermission("SuperAdmin")]
[ModifyPermission("SuperAdmin")]
public sealed partial class TiposDocumentoRow : Row<TiposDocumentoRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Tipo Documento"), Identity, IdProperty]
        public int? IdTipoDocumento { get; set; }

        [DisplayName("Nombre Tipo Documento"), Size(200), NotNull, QuickSearch, NameProperty]
        public string NombreTipoDocumento { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }

        [DisplayName("Sigla"), Size(50), NotNull]
        public string Sigla { get; set; }
    }
}